resources/js/Components/CustomFieldRuleBuilder.tsx

```
@@ -0,0 +1,312 @@
+                                </Select.Option>
+                                <Select.Option value="OR">
+                                    OR (any criterion must match)
+                                </Select.Option>
+                            </Select>
+                        </Form.Item>
+
+                        <Form.List name="criteria">
+                            {(fields, { add, remove }) => (
+                                <>
+                                    {fields.map((fieldItem, index) => {
+                                        const operator = Form.useWatch(
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)